ibis/expr/operations/relations.py

```diff
@@ -310,14 +310,14 @@ class AnyLeftJoin(Join):
 
 @public
 class LeftSemiJoin(Join):
-    @property
+    @attribute
```

Comment:
For simple properties we shouldn't use attribute, and it is not contributing to the call stack, see my explanation below.

Reply:
Think about a deeply nested semi join. What would the `@property` behavior be?

Reply:
1000 is the usual recursion limit, this would be triggered after 1000 nested joins.

Reply:
I'm not totally against it since it has a negligible effect, but I'm not sure what expectations we have against our IR representation.

Reply:
We're hitting the limit somewhere between 600 and 800 relations right now I believe.